core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -134,4 +134,7 @@ private TableProperties() {
 
   public static final String ENGINE_HIVE_ENABLED = "engine.hive.enabled";
   public static final boolean ENGINE_HIVE_ENABLED_DEFAULT = false;
+
+  public static final String GC_ENABLED = "gc.enabled";
+  public static final boolean GC_ENABLED_DEFAULT = true;
```

Comment:
I like this. We can also add a property to require the action is used instead of the table operation to ensure that we use reachability instead of incremental changes.